pkg/routeagent/controllers/route/driver.go

```diff
@@ -3,32 +3,59 @@ package route
 import (
 	"fmt"
 	"io/ioutil"
+	"net"
 
-	"github.com/vishvananda/netlink"
 	"k8s.io/klog"
 )
 
-const FlannelInterface = "flannel.1"
+func discoverCNIInterface(clusterCIDR string) *cniInterface {
```

Comment:
Should this function propagate errors? And does an error here warrant aborting start up?

Reply:
I'm afraid No. We are only trying to discover the CNI interface in this API. If it does not exist (like some new CNI), its fine. We should not abort the startup.